src/AggregateFunctions/AggregateFunctionGroupArraySorted.h

```diff
@@ -203,16 +203,16 @@ class AggregateFunctionGroupArraySorted : public IAggregateFunctionDataHelper<
 
     template <typename TColumn, bool is_plain, typename TFunc>
     void
-    forFirstRows(size_t batch_size, const IColumn ** columns, size_t data_column, Arena * arena, ssize_t if_argument_pos, TFunc func) const
+    forFirstRows(size_t row_begin, size_t row_end, const IColumn ** columns, size_t data_column, Arena * arena, ssize_t if_argument_pos, TFunc func) const
     {
         const TColumn * values = nullptr;
         std::unique_ptr<std::vector<TColumn>> values_vector;
         std::vector<size_t> best_rows(threshold);
 
         if constexpr (std::is_same_v<TColumn, StringRef>)
         {
-            values_vector.reset(new std::vector<TColumn>(batch_size));
-            for (size_t i = 0; i < batch_size; i++)
+            values_vector.reset(new std::vector<TColumn>(row_end));
```

Comment:
>Is it possible to allocate in similar code row_end - row_begin instead of row_end?

Yes, but this interface is similar to Aggregator interface, which has overhead for `row_begin`, since avoiding this will require changing the interface, and I want to make this thing separately (to avoid constant conflicts)